cli/mender-artifact/modify.go

```diff
@@ -41,61 +46,40 @@ func modifyArtifact(c *cli.Context) error {
 		return cli.NewExitError("File ["+c.Args().First()+"] does not exist.", 1)
 	}
 
-	candidateType, modifyCandidates, err :=
-		getCandidatesForModify(c.Args().First())
+	image, err := virtualImage.Open(comp, privateKey, c.Args().First())
 
 	if err != nil {
 		return cli.NewExitError("Error selecting images for modification: "+err.Error(), 1)
 	}
-	// strip the data and boot partitions
-	if candidateType == RootfsImageArtifact {
-		modifyCandidates = modifyCandidates[0:1]
-		for _, mc := range modifyCandidates {
-			defer os.Remove(mc.path)
-		}
-	} else if candidateType == RawSDImage {
-		if len(modifyCandidates) == 4 { // sdimg
-			defer func(partitions []partition) {
-				for _, part := range partitions {
-					os.Remove(part.path)
-				}
-			}(modifyCandidates)
-			modifyCandidates = modifyCandidates[1:3]
+	defer func() {
+		if err == nil {
+			err = image.Close()
+		} else {
+			image.Close()
 		}
-	}
+	}()
 
-	for _, toModify := range modifyCandidates {
-		if err := modifyExisting(c, toModify.path); err != nil {
-			return cli.NewExitError("Error modifying artifact["+toModify.path+"]: "+
-				err.Error(), 1)
-		}
+	if err := modifyExisting(c, image); err != nil {
+		return cli.NewExitError("Error modifying artifact["+c.Args().First()+"]: "+
+			err.Error(), 1)
 	}
 
-	if len(modifyCandidates) == 2 {
-		// make modified images part of sdimg again
-		if err := repackSdimg(modifyCandidates, c.Args().First()); err != nil {
-			return cli.NewExitError("Can not recreate sdimg file: "+err.Error(), 1)
-		}
-		return nil
-	}
-
-	if candidateType == RootfsImageArtifact || candidateType == ModuleImageArtifact {
-		// re-create the artifact
-		rootfs := ""
-		if modifyCandidates != nil {
-			rootfs = modifyCandidates[0].path
-		}
-		err := repackArtifact(comp, c.Args().First(), rootfs,
-			c.String("name"))
-		if err != nil {
-			return cli.NewExitError("Can not recreate artifact: "+err.Error(), 1)
-		}
-	}
 	return nil
 }
 
 // oblivious to whether the file exists beforehand
-func modifyName(name, image string) error {
+func modifyArtifactInfoName(name string, image VPImage) error {
+	attributeModified := false
+
+	if art, ok := image.(*ModImageArtifact); ok {
```

Comment:
Ref your commit message: I see no issue with discriminating on a type like this, in fact I like it!